Std/Data/Nat/Lemmas.lean

```diff
@@ -99,63 +100,55 @@ theorem succ_le {n m : Nat} : succ n ≤ m ↔ n < m := .rfl
 theorem lt_succ {m n : Nat} : m < succ n ↔ m ≤ n :=
   ⟨le_of_lt_succ, lt_succ_of_le⟩
 
-theorem lt_succ_of_lt (h : a < b) : a < succ b := le_succ_of_le h
+theorem lt_succ_of_lt {a b : Nat} (h : a < b) : a < succ b := le_succ_of_le h
 
-theorem succ_ne_self : ∀ n : Nat, succ n ≠ n
-  | 0,   h => absurd h (succ_ne_zero 0)
-  | n+1, h => succ_ne_self n (Nat.noConfusion h id)
+theorem succ_ne_self : ∀ (n : Nat), succ n ≠ n
+  | _+1, h => succ_ne_self _ (succ.inj h)
 
 theorem succ_pred_eq_of_pos : ∀ {n : Nat}, 0 < n → succ (pred n) = n
-  | succ _, _ => rfl
+  | _+1, _ => rfl
 
-theorem eq_zero_or_eq_succ_pred (n : Nat) : n = 0 ∨ n = succ (pred n) := by
-  cases n <;> simp
+theorem eq_zero_or_eq_succ_pred : ∀ (n : Nat), n = 0 ∨ n = succ (pred n)
+  | 0 => .inl rfl
+  | _+1 => .inr rfl
 
-theorem exists_eq_succ_of_ne_zero {n : Nat} (H : n ≠ 0) : ∃ k, n = succ k :=
-  ⟨_, (eq_zero_or_eq_succ_pred _).resolve_left H⟩
+theorem exists_eq_succ_of_ne_zero : ∀ {n : Nat}, n ≠ 0 → ∃ k, n = succ k
+  | _+1, _ => ⟨_, rfl⟩
 
 theorem succ_eq_one_add (n : Nat) : n.succ = 1 + n := by
   rw [Nat.succ_eq_add_one, Nat.add_comm]
 
-theorem succ_inj' : succ n = succ m ↔ n = m :=
+theorem succ_inj' {n m : Nat} : succ n = succ m ↔ n = m :=
```

Comment:
What is the reason for this change? Style-wise, std prefers auto-implicits, although it's fine to include types if you think it is clearer or you need to manipulate argument order. I would prefer to leave them off in most cases here, it's pretty obvious everything is about Nat.

Reply:
Easy fix!

Reply:
Should I do the reverse and remove implicit parameters that could be auto-implicit?

Reply:
That's fine by me. Sometimes it makes a small difference with order of parameters, but I think we want to normalize to the ordering you get with autoimplicits anyway.